src/masternodeman.cpp

```diff
@@ -776,3 +836,109 @@ void CMasternodeMan::UpdateLastPaid(const CBlockIndex *pindex) {
     // every time is like the first time if winners list is not synced
     IsFirstRun = !masternodeSync.IsWinnersListSynced();
 }
+
+void CMasternodeMan::UpdateWatchdogVoteTime(const CTxIn& vin)
+{
+    LOCK(cs);
+    CMasternode* pMN = Find(vin);
+    if(!pMN)  {
+        return;
+    }
+    pMN->UpdateWatchdogVoteTime();
+    nLastWatchdogVoteTime = GetTime();
+}
+
+bool CMasternodeMan::IsWatchdogActive()
+{
+    LOCK(cs);
+    // Check if any masternodes have voted recently, otherwise return false
+    if((GetTime() - nLastWatchdogVoteTime) > MASTERNODE_WATCHDOG_MAX_SECONDS) {
```

Comment:
Can be simplified to `return GetTime() - nLastWatchdogVoteTime < MASTERNODE_WATCHDOG_MAX_SECONDS;`